src/NexusMods.DataModel/ArchiveInstaller.cs

```diff
@@ -49,21 +47,15 @@ public class ArchiveInstaller : IArchiveInstaller
     }
     
     /// <inheritdoc />
-    public async Task<ModId[]> AddMods(LoadoutId loadoutId, DownloadAnalysis.Model download, string? defaultModName = null, IModInstaller? installer = null, CancellationToken token = default)
+    public async Task<ModId[]> AddMods(LoadoutId loadoutId, DownloadAnalysis.Model download, IModInstaller? installer = null, CancellationToken token = default)
     {
         // Get the loadout and create the mod, so we can use it in the job.
         var useCustomInstaller = installer != null;
         var loadout = _conn.Db.Get<Loadout.Model>(loadoutId.Value);
 
         var archiveName = download.TryGet(FilePathMetadata.OriginalName, out var originalName) ? originalName.ToString() : null;
         var suggestedName = download.TryGet(DownloadAnalysis.SuggestedName, out var outSuggestedName) ? outSuggestedName : null;
-
-        var modName = defaultModName ?? suggestedName ?? archiveName ?? "<unknown>";
-        {
-            using var dlTx = _conn.BeginTransaction();
-            dlTx.Add(download.Id, DownloadAnalysis.SuggestedName, modName);
-            await dlTx.Commit();
-        }
+        var modName = suggestedName ?? archiveName ?? "Unknown";
```

Comment:
When will `suggestedName` be non-null? Can it be non-null after this PR?

Reply:
> When will `suggestedName` be null?

Seems like you're onto something, yeah, this can be improved.

Originally I was under the impression downloads from 0.5 (e.g. HTTP Downloads) would have exhibited the edge case, if they were downloaded but not added. But after doublechecking, that's not true, let me improve this. 

> Can it be null after this PR?

It cannot